packages/builders/src/components/button/Button.ts

```diff
@@ -93,8 +93,8 @@ export class ButtonBuilder extends ComponentBuilder<APIButtonComponent> {
 	 *
 	 * @param emoji - The emoji to display on this button
 	 */
-	public setEmoji(emoji: APIMessageComponentEmoji) {
-		this.data.emoji = emojiValidator.parse(emoji);
+	public setEmoji(emoji: APIMessageComponentEmoji | null) {
+		this.data.emoji = emojiValidator.nullish.parse(emoji) ?? undefined;
```

Comment:
- shouldn't this (and others) use `nullable` instead of `nullish`?
 - doesn't this create a new validator each time the method is being run?

Reply:
> * shouldn't this (and others) use `nullable` instead of `nullish`?

That would allow `undefined`, which I didn't do for consistency with `EmbedBuilder`.


> * doesn't this create a new validator each time the method is being run?

Fixed.

Reply:
- [nullish](https://www.sapphirejs.dev/docs/Documentation/api-shapeshift/classes/BaseValidator#nullish) accepts `undefined`, [nullable](https://www.sapphirejs.dev/docs/Documentation/api-shapeshift/classes/BaseValidator#nullable) does not
- `setDescription` needs its own (nullable) validator too